### ISSUE TITLE
Questions about training & testing

### ISSUE THREAD

**Issue by codeslake** [shallowdream204/SODA-SR#10]
Hi, thanks for fabulous work!

### Questions about Training:
- According to the [G_optimizer_reuse](https://github.com/shallowdream204/SODA-SR/blob/main/options/swinir/train_soda_sr_classical.json#L67) in the config and [this line](https://github.com/shallowdream204/SODA-SR/blob/main/train_test_scripts/main_train_sodasr.py#L72) in the SODA-based training script, it seems that you [load the pretrained optimizer](https://github.com/shallowdream204/SODA-SR/blob/main/models/model_plain.py#L70) for SODA training. Is the `G_optimizer_reuse` flag used for resuming SODA training, or is it enabled from the start of the SODA training process? If it’s the latter, how do you load the optimizer that was used for pretraining SwinIR without WAT? The optimizer for SODA training is designed for SwinIR with WAT, which seems like it could cause a conflict in the network architecture.

- As far as I understand, `model_plain.py` outlines the SODA training scheme, but it doesn't seem to include Equations (10, 11) from the paper. Could you share the model file that contains the complete loss functions mentioned in the paper?

- Can you please share training config / code for CDC network / model file for training the CDC network with the SODA scheme?

### Questions about Testing:
- Could you share the checkpoints and network code for the final models (the 6 SFSR cases) reported in the paper? I’m looking for them to reproduce both the quantitative and qualitative results.

- From what I gather in the paper, the quantitative experiments seem to use the CDC backbone architecture, while the qualitative ones use the SwinIR backbone. I found the SODA-based SwinIR architecture in the repository but couldn’t find the CDC network. Could you share the code for the SODA-based CDC? I’m particularly curious about how WAT is applied for the student and how Gumbel-Softmax is used for the teacher.

**Comment by shallowdream204** [comment]
We did not reuse the G_optimizer, which might be the default setting in KAIR. In fact, we did not load any pretrained optimizer.

We have uploaded the pre-trained  SwinIR weights in [huggingface](https://huggingface.co/shallowdream204/SODA-SR/tree/main), you can refer to [here](https://github.com/shallowdream204/SODA-SR/tree/main?tab=readme-ov-file#evaluation) and use the updated `main_test_sodasr.py` to evaluate all the models.

I'm sorry that I've been busy with other projects lately. More code related to CDC and wavelet-based GAN loss will be organized and released later. I apologize for any inconvenience this may have caused you.

**Comment by codeslake** [comment]
Thanks, @shallowdream204!

I have a few follow up questions though,

- Please share reproducible training code of the proposed self-training scheme.
  - I've been trying to reproduce the quantitative table of the paper (both CVPR official (CDC) and arXiv v1,version (SwinIR)). However, I cannot reproduce the numbers. Note that for SwinIR, I've been using the exact same code (`main_train_sodasr.py`) uploaded in the repo. Also, please note that I've also tried experimenting models with Eqs. 10 and 11.
    - I've reproduced "source-only" models (target-only rows for each column) for each CDC and SwinIR, but I could not reproduce numbers for SODA-SRed models.
    - Whenever I enable WAT (as in the last column of Table 3), the final PSNR is 30.73dB for CDC and 30.06dB for SwinIR. When I disable WAT (as in 3rd column in Table 3), I get higher PSNR, 30.94dB for CDC and 29.85dB for SwinIR, which are far lower that what you guys report in the paper.
    - Sharing final check points really helps, but I'd like to ask authors to kindly share the main contribution of the paper, the source-free self-training scheme with WAT. Can you guys please share reproducible training code?

- About CDC, if it is hard for you guys to share the code at the moment, can you please tell where exactly Gumbel softmax is applied in the architecture, and WAT is applied?

**Comment by Renzhihan** [comment]
What does "_bi" mean, such as "p2o_bi.pth" in the huggingface. Meanwhile, there is no o2s.pth. Could you upload it?
